### PR TITLE
fix(sdk.v2): fix a small bug in `io_types.is_artifact_annotation()`

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Before the fix, `io_types.is_artifact_annotation(Optional[str])` would return `True`.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/dsl/io_types_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 The Kubeflow Authors
```
Comment (neuromage) [inline sdk/python/kfp/dsl/io_types_test.py:1]: Add it to existing test file here?
https://github.com/kubeflow/pipelines/blob/master/sdk/python/tests/dsl/io_types_test.py

Comment (chensun) [reply sdk/python/kfp/dsl/io_types_test.py:1]: I see. I didn't notice this exists. 
Shouldn't we move that one here? We were trying to adopt the style that unittests files are next to the source files (at least for new tests and sources we're adding).

Comment (neuromage) [reply sdk/python/kfp/dsl/io_types_test.py:1]: Yeah, moving that here is fine with me as well. Do you mind doing that in this PR?

Comment (chensun) [reply sdk/python/kfp/dsl/io_types_test.py]: Done. Also moved `metrics_utils_test.py`.

**Pull request merged**